lib/resty/jwt.lua

```diff
@@ -549,7 +551,7 @@ function _M.sign(self, secret_key, jwt_obj)
     end
   end
 
-  if typ == str_const.JWE or jwt_obj.header.enc then
+  if jwt_obj.typ == str_const.JWE or (jwt_obj.typ == nil and (typ == str_const.JWE or jwt_obj.header.enc)) then
```

Comment:
Does the jwt_obj have `typ`?

Reply:
<img width="725" alt="image" src="https://github.com/api7/lua-resty-jwt/assets/2706161/f0f3eca2-765d-4ab7-865d-9a744df9b4ee">

Reply:
The jwt_obj is not from the function `parse_jwt`

Reply:
![image](https://github.com/api7/lua-resty-jwt/assets/9354193/faeba915-9022-458b-bebe-66ddd71b2299)
It's a jwt table we construct

Reply:
![image](https://github.com/api7/lua-resty-jwt/assets/2706161/c2929d4f-a5f7-45ca-ab46-d880597cb745)
So it is necessary to write this way to consider the use of compatible old APIs

Reply:
![image](https://github.com/api7/lua-resty-jwt/assets/2706161/bf723adc-4b01-4bab-bf33-0fcb8d3486c5)